File: backend/models/member.go
package models

import "time"

// MemberLevel 会员等级
type MemberLevel string

const (
	Regular  MemberLevel = "regular"  // 普通会员
	Silver   MemberLevel = "silver"   // 白银会员
	Gold     MemberLevel = "gold"     // 黄金会员
	Platinum MemberLevel = "platinum" // 铂金会员
	Diamond  MemberLevel = "diamond"  // 钻石会员
)

// StylePreference 风格偏好
type StylePreference string

const (
	Casual     StylePreference = "casual"     // 休闲
	Formal     StylePreference = "formal"     // 正式
	Sportswear StylePreference = "sportswear" // 运动
	Vintage    StylePreference = "vintage"    // 复古
	Minimalist StylePreference = "minimalist" // 极简
	Romantic   StylePreference = "romantic"   // 浪漫
	Bohemian   StylePreference = "bohemian"   // 波西米亚
	Street     StylePreference = "street"     // 街头
)

// ConsumptionLevel 消费能力
type ConsumptionLevel string

const (
	Low    ConsumptionLevel = "low"    // 低
	Medium ConsumptionLevel = "medium" // 中
	High   ConsumptionLevel = "high"   // 高
	Luxury ConsumptionLevel = "luxury" // 奢侈
)

type Member struct {
	ID               uint   `gorm:"primaryKey"`
	OpenID           string `gorm:"size:50;uniqueIndex"` // 微信 OpenID
	Name             string `gorm:"size:50;not null"`
	Phone            string `gorm:"size:20;uniqueIndex"`
	Gender           string `gorm:"size:10"`
	Birthday         *time.Time
	Email            string      `gorm:"size:100"`
	Address          string      `gorm:"size:255"`
	Level            MemberLevel `gorm:"size:20;default:'regular'"`
	Points           int         `gorm:"default:0"` // 积分
	TotalSpent       float64     `gorm:"default:0"` // 累计消费金额
	LastPurchaseDate *time.Time

	// 体型数据
	BodyHeight    int // 身高(cm)
	BodyWeight    int // 体重(kg)
	ShoulderWidth int // 肩宽(cm)
	BustSize      int // 胸围(cm)
	WaistSize     int // 腰围(cm)
	HipSize       int // 臀围(cm)
	Inseam        int // 内缝长度(cm)

	// 偏好数据
	StylePreference    StylePreference  `gorm:"size:20"`  // 风格偏好
	FavoriteColors     string           `gorm:"size:100"` // 喜欢的颜色，逗号分隔
	FavoriteCategories string           `gorm:"size:100"` // 喜欢的品类，逗号分隔
	ConsumptionLevel   ConsumptionLevel `gorm:"size:20"`  // 消费能力

	Note      string `gorm:"size:255"` // 备注
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: mobile/lib/models/member.dart
class Member {
  final int id;
  final String name;
  final String phone;
  final String gender;
  final String? birthday;
  final String email;
  final String address;
  final String level;
  final int points;
  final double totalSpent;
  final String? lastPurchaseDate;
  
  // 体型数据
  final int? bodyHeight;
  final int? bodyWeight;
  final int? shoulderWidth;
  final int? bustSize;
  final int? waistSize;
  final int? hipSize;
  final int? inseam;
  
  // 偏好数据
  final String? stylePreference;
  final String? favoriteColors;
  final String? favoriteCategories;
  final String? consumptionLevel;
  
  final String note;
  final String createdAt;
  final String updatedAt;
  
  Member({
    required this.id,
    required this.name,
    required this.phone,
    required this.gender,
    this.birthday,
    required this.email,
    required this.address,
    required this.level,
    required this.points,
    required this.totalSpent,
    this.lastPurchaseDate,
    this.bodyHeight,
    this.bodyWeight,
    this.shoulderWidth,
    this.bustSize,
    this.waistSize,
    this.hipSize,
    this.inseam,
    this.stylePreference,
    this.favoriteColors,
    this.favoriteCategories,
    this.consumptionLevel,
    required this.note,
    required this.createdAt,
    required this.updatedAt,
  });
  
  factory Member.fromJson(Map<String, dynamic> json) {
    return Member(
      id: json['ID'],
      name: json['Name'],
      phone: json['Phone'],
      gender: json['Gender'] ?? '',
      birthday: json['Birthday'],
      email: json['Email'] ?? '',
      address: json['Address'] ?? '',
      level: json['Level'],
      points: json['Points'] ?? 0,
      totalSpent: json['TotalSpent']?.toDouble() ?? 0.0,
      lastPurchaseDate: json['LastPurchaseDate'],
      bodyHeight: json['BodyHeight'],
      bodyWeight: json['BodyWeight'],
      shoulderWidth: json['ShoulderWidth'],
      bustSize: json['BustSize'],
      waistSize: json['WaistSize'],
      hipSize: json['HipSize'],
      inseam: json['Inseam'],
      stylePreference: json['StylePreference'],
      favoriteColors: json['FavoriteColors'],
      favoriteCategories: json['FavoriteCategories'],
      consumptionLevel: json['ConsumptionLevel'],
      note: json['Note'] ?? '',
      createdAt: json['CreatedAt'],
      updatedAt: json['UpdatedAt'],
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'ID': id,
      'Name': name,
      'Phone': phone,
      'Gender': gender,
      'Birthday': birthday,
      'Email': email,
      'Address': address,
      'Level': level,
      'Points': points,
      'TotalSpent': totalSpent,
      'LastPurchaseDate': lastPurchaseDate,
      'BodyHeight': bodyHeight,
      'BodyWeight': bodyWeight,
      'ShoulderWidth': shoulderWidth,
      'BustSize': bustSize,
      'WaistSize': waistSize,
      'HipSize': hipSize,
      'Inseam': inseam,
      'StylePreference': stylePreference,
      'FavoriteColors': favoriteColors,
      'FavoriteCategories': favoriteCategories,
      'ConsumptionLevel': consumptionLevel,
      'Note': note,
      'CreatedAt': createdAt,
      'UpdatedAt': updatedAt,
    };
  }
}
